pkg/azuremachinepool/storageaccounttype.go

```diff
@@ -0,0 +1,36 @@
+package azuremachinepool
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
+	"github.com/giantswarm/microerror"
+	expcapzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/internal/vmcapabilities"
+)
+
+func checkStorageAccountTypeIsValid(ctx context.Context, vmcaps *vmcapabilities.VMSKU, azureMachinePool *expcapzv1alpha3.AzureMachinePool) error {
+	selectedStorageAccount := azureMachinePool.Spec.Template.OSDisk.ManagedDisk.StorageAccountType
+
+	if selectedStorageAccount != string(compute.StorageAccountTypesStandardLRS) &&
+		selectedStorageAccount != string(compute.StorageAccountTypesPremiumLRS) {
+		// Storage account type is invalid.
+		return microerror.Maskf(invalidOperationError, "Storage account type '%s' is invalid. Allowed values are %s and %s", selectedStorageAccount, string(compute.StorageAccountTypesStandardLRS), string(compute.StorageAccountTypesPremiumLRS))
```

Comment:
```suggestion
		return microerror.Maskf(invalidOperationError, "Storage account type %q is invalid. Allowed values are %q and %q", selectedStorageAccount, string(compute.StorageAccountTypesStandardLRS), string(compute.StorageAccountTypesPremiumLRS))
```

> %q	a double-quoted string safely escaped with Go syntax

That way it should read something like following:
`Storage account type "foobarLRS" is invalid. Allowed values are "StandardLRS" and "PremiumLRS"`